src/main/java/org/saltyrtc/tasks/webrtc/WebRTCTask.java

```diff
@@ -421,6 +420,14 @@ public void onStateChange() {
                 final SignalingInterface signaling = WebRTCTask.this.signaling;
                 final SecureDataChannel sdc = WebRTCTask.this.sdc;
 
+                // When the close() method is called before this state change callback is triggered,
+                // the SecureDataChannel instance may be null.
+                // In that case, ignore the state change.
+                if (sdc == null) {
+                    logger.debug("DataChannel: onStateChange, but data channel is null");
```

Comment:
Can't we unbind the event handler instead of doing this? Unbinding events in general is a good idea when closing.

Reply:
How would you unbind an event on `this.sdc` if `this.sdc` is null? :slightly_smiling_face: The only way would be to enclose a reference to the SecureDataChannel (as described on the other comment), but that doesn't feel clean to me.

Reply:
I'm not suggesting to enclose a reference to the data channel. Just to unbind all data channel events before setting `sdc` to `null`.

Reply:
I updated the PR, the observer is now unbound.